Details book screen

SoleCincis opened the pull request:
Current status of Detail screen :
![IMG_1916](https://user-images.githubusercontent.com/43641321/141455466-5b43c40f-1885-443c-aa98-7c73ffd4d47f.PNG) ![IMG_712A1B2CCDDF-1](https://user-images.githubusercontent.com/43641321/141462107-ea832815-9f7d-43a5-8c21-79dc2ab9ffa4.jpeg)

SoleCincis committed:
feature(app.js): new stack screen Detail

- and new DetailScreen component showing a title and a subtitle

SoleCincis committed:
Feat(booksScreen): passing data as params

- first attempt to get data into detail screen failed

SoleCincis committed:
chore: rebased this branch on books API request

- lost TouchableOpacity during rebase, then I added it again, that's way it counts as change
- deleted "mode modal" because effect of screen coming from the bottom feels unnatural if navigation starts from a list entry

SoleCincis committed:
feat: show data in detail screen

- passing navigation and item as props from FlatList component to RenderItem component
- passing item as params to detailScreen

SoleCincis commented on screens/DetailScreen.js:
```diff
@@ -0,0 +1,31 @@
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>{title}</Text>
+      <Text style={styles.subtitle}>{author.length > 1 ? `Authors: ${author}, ` : `Author: ${author}`}</Text>
```
I haven't jet figured out a nice way to identify last item of authors array which should not have a comma . 
Maybe something like :  
`{author.length > 1  ? Authors: ${author},  
?  author.length == author.length -1  ? Author: ${author} 
: Author: ${author} }`

digorath replied:
1. think if there is a better name for the `author` variable
2. `author.length == author.length -1` will always be false
2. why does it even work as it is at the moment?
3. take a look at [Array.prototype.join()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join)

SoleCincis replied:
thanks , I don't know why I did that.. fixed in f4239d901e05b42a1e30b540408b9b09ab87650d

1.  author is now listOfAuthors
2. indeed non sense idea 
3.  I guess it works because my params is authors, so author works but it dosen't make sense logically to be called like that 
4.  very useful

digorath replied:
about 3.: i was more thinking of you implicitly using the [Array.prototype.toString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toString) method

digorath commented:
there are commits in this PR, that should not be here.

SoleCincis committed:
feat(DetailScreen): fix on authors list

- comma is part of the array then obviously no need to print it
- listOfAuthors using join(', ')  for concatenating all the elements of the array with a comma and space
- authors has to be there as well when index is 1
